framework/src/auxkernels/ParsedVectorAux.C

```diff
@@ -98,11 +91,19 @@ ParsedVectorAux::ParsedVectorAux(const InputParameters & parameters)
     setParserFeatureFlags(_func_F[i]);
 
     // add the constant expressions
-    if (getParam<std::vector<std::vector<std::string>>>("constant_names").size())
-      addFParserConstants(
-          _func_F[i],
-          getParam<std::vector<std::vector<std::string>>>("constant_names")[i],
-          getParam<std::vector<std::vector<std::string>>>("constant_expressions")[i]);
+    auto constant_names = isParamValid("constant_names")
+                              ? getParam<std::vector<std::vector<std::string>>>("constant_names")
+                              : std::vector<std::vector<std::string>>{};
+    auto constant_expressions =
+        isParamValid("constant_expressions")
+            ? getParam<std::vector<std::vector<std::string>>>("constant_expressions")
+            : std::vector<std::vector<std::string>>{};
+
+    if (constant_names.size() != constant_expressions.size())
+      paramError("constant_names", "Must be same length as constant_expressions");
+
+    if (isParamValid("constant_names") && isParamValid("constant_expressions"))
+      addFParserConstants(_func_F[i], constant_names[i], constant_expressions[i]);
```

Comment:
why is anything changing in this file?
the previous code was fine, you could just try to use {} instead of the defaults they had

Reply:
This `isParamValid` is used to replace `getParam<std::vector<std::vector<std::string>>>("constant_names").size()`. It appears multiple times in the conditions which is not necessary. `IsParamValid` is cheaper than calling 'getParam.size()'.

Reply:
you can restore the default and then use isParamSetByUser instead of isParamValid. This lets you change a single line instead of rewriting the entire constuctor

Reply:
It doesn't do any harm to the constructor or failed any test cases. Just different ways of writing it. Why this can not be accepted?

Reply:
revert back

Reply:
I don't think it is necessary to check size() multiple times

Reply:
This could be made more elegant with `getParam<std::string, std::string>("constant_names", "constant_expressions")` which gets both parameters as a vector of pairs, checking that they have the same length. (with the proper change in `addFParserConstants`). Don't address this in this PR. I can make that change later.